app/src/main/kotlin/at/bitfire/davdroid/ui/composable/BasicTopAppBar.kt

```diff
@@ -0,0 +1,34 @@
+package at.bitfire.davdroid.ui.composable
+
+import android.app.Activity
+import android.content.Intent
+import androidx.annotation.StringRes
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import at.bitfire.davdroid.R
+import kotlin.reflect.KClass
+
+@Composable
+fun Activity.BasicTopAppBar(@StringRes titleStringRes: Int, parentActivity: KClass<*>) {
+    TopAppBar(
+        title = { Text(stringResource(titleStringRes)) },
+        navigationIcon = {
+            IconButton(
+                onClick = {
+                    // Finish the current activity
+                    finish()
+                    // And start the parent one
+                    startActivity(Intent(this, parentActivity.java))
```

Comment:
We should use `AppCompatActivity.onSupportNavigateUp()` instead for [upward navigation](https://m2.material.io/design/navigation/understanding-navigation.html#reverse-navigation).

`finish` + `startActivity` could manipulate the recent tasks etc in a way that we don't want.

We can then also omit the `parentActivity` argument, which is taken from the Manifest by `onSupportNavigateUp`.